plugins/synapticsmst/fu-plugin-synapticsmst.c

```diff
@@ -39,468 +42,109 @@ fu_synapticsmst_check_amdgpu_safe (GError **error)
 	return TRUE;
 }
 
-static gboolean
-fu_synapticsmst_check_supported_system (FuPlugin *plugin, GError **error)
-{
-
-	if (g_getenv ("FWUPD_SYNAPTICSMST_FW_DIR") != NULL) {
-		g_debug ("Running Synaptics plugin in test mode");
-		return TRUE;
-	}
-
-	/* See https://github.com/hughsie/fwupd/issues/1121 for more details */
-	if (!fu_synapticsmst_check_amdgpu_safe (error))
-		return FALSE;
-
-	if (!g_file_test (SYSFS_DRM_DP_AUX, G_FILE_TEST_IS_DIR)) {
-		g_set_error (error,
-			     G_IO_ERROR,
-			     G_IO_ERROR_INVALID_DATA,
-			     "MST firmware updating not supported, missing kernel support.");
-		return FALSE;
-	}
-	return TRUE;
-}
-
-/* creates MST-$str-$BOARDID */
-static void
-fu_plugin_synapticsmst_create_simple_guid (FuDevice *fu_device,
-					   FuSynapticsmstDevice *device,
-					   const gchar *str)
-{
-	guint16 board_id = fu_synapticsmst_device_get_board_id (device);
-	g_autofree gchar *devid = g_strdup_printf ("MST-%s-%u", str, board_id);
-	fu_device_add_instance_id (fu_device, devid);
-}
-
-/* creates MST-$str-$chipid-$BOARDID */
-static void
-fu_plugin_synapticsmst_create_complex_guid (FuDevice *fu_device,
-					    FuSynapticsmstDevice *device,
-					    const gchar *device_kind)
-{
-	const gchar *chip_id_str = fu_synapticsmst_device_get_chip_id_str (device);
-	g_autofree gchar *chip_id_down = g_ascii_strdown (chip_id_str, -1);
-	g_autofree gchar *tmp = g_strdup_printf ("%s-%s", device_kind, chip_id_down);
-
-	fu_plugin_synapticsmst_create_simple_guid (fu_device, device, tmp);
-}
-
-static gboolean
-fu_plugin_synapticsmst_lookup_device (FuPlugin *plugin,
-				      FuDevice *fu_device,
-				      FuSynapticsmstDevice *device,
-				      GError **error)
+gboolean
+fu_plugin_udev_device_added (FuPlugin *plugin, FuUdevDevice *device, GError **error)
 {
-	const gchar *board_str;
-	const gchar *guid_template;
-	guint16 board_id = fu_synapticsmst_device_get_board_id (device);
-	const gchar *chip_id_str = fu_synapticsmst_device_get_chip_id_str (device);
-	g_autofree gchar *group = NULL;
-	g_autofree gchar *name = NULL;
+	FuPluginData *priv = fu_plugin_get_data (plugin);
+	g_autoptr(FuDeviceLocker) locker = NULL;
+	g_autoptr(FuSynapticsmstDevice) dev = NULL;
+	g_autoptr(GError) error_local = NULL;
 
-	/* GUIDs used only for test mode */
-	if (g_getenv ("FWUPD_SYNAPTICSMST_FW_DIR") != NULL) {
-		g_autofree gchar *tmp = NULL;
-		tmp = g_strdup_printf ("test-%s", chip_id_str);
-		fu_plugin_synapticsmst_create_simple_guid (fu_device, device, tmp);
+	/* interesting device? */
+	if (g_strcmp0 (fu_udev_device_get_subsystem (device), "drm_dp_aux_dev") != 0)
```

Comment:
We use the same function during coldplug at startup, i'm just covering my bases in case that a `drm_dp_aux_dev` class device somehow gets created at runtime.